Automattic/wp-calypso
# Plans (State): Update post-share component to utilise Plans data store

Part of addressing https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

- Updates post-share component to utilize Plans data store for pricing/currency and current plan
- Other cleanup - updates component to TS, fixes props/imports, etc.

### Media

<img width="600" alt="Screenshot 2024-07-22 at 12 43 00 PM" src="https://github.com/user-attachments/assets/9c95df43-2f94-4084-bedf-51d2df32de37">

## Why are these changes being made?


We only need one form of handling plan pricing in the code base, more so one framework. Anything else risks having alternative and differing views of the same data propagating across the code base.

## Testing Instructions



* On a site with a **Free** plan, go to `/posts/[ site ]`
* Click on settings and then "Share"
* Connect a social account e.g. Tumblr, if not already
* Confirm the upsell to Premium nudge renders correctly with right info (pricing, currency, features)
   * If site includes upgrade credits, then prorated prices should be shown, as per the original

## Pre-merge Checklist
